src/dlfcn.c

```diff
@@ -474,7 +479,207 @@ char *dlerror( void )
     return error_buffer;
 }
 
-#ifdef SHARED
+/* see https://docs.microsoft.com/en-us/archive/msdn-magazine/2002/march/inside-windows-an-in-depth-look-into-the-win32-portable-executable-file-format-part-2
+ * for details */
+
+/*
+ * Return pointer to import directory
+ */
+static IMAGE_IMPORT_DESCRIPTOR* get_import_directory( HMODULE module, DWORD *size )

```

Comment:
Half of this `get_import_directory` function is same as `get_import_address_table`. What about joining these two functions into one?

```c
static void parse_image_header(HMODULE module, IMAGE_IMPORT_DESCRIPTOR **iid, DWORD *iidSize, void **iat, DWORD *iatSize);
```